src/users/users.service.ts

```diff
@@ -1,35 +1,69 @@
 import { Injectable } from '@nestjs/common';
 import { CreateUserDto } from './dto/create-user.dto';
-import { SetSshDto } from './dto/set-ssh.dto';
+import { SetSshDto } from 'src/users/dto/set-ssh.dto';
 import { PrismaService } from 'src/prisma.service';
 import { User, Prisma } from '@prisma/client';
 import { SanitizedUser } from './types/sanitized-user.type';
 import { exclude } from 'src/utils/prisma-exclude';
+import * as bcrypt from 'bcrypt';
+import { EMAIL_REGEX, PASSWORD_REGEX, USERNAME_REGEX } from 'src/utils/constants';
 
 @Injectable()
 export class UsersService {
   constructor(private prisma: PrismaService) {}
 
-  private passwd_encrypt(password: string) {
-    // FIXME: implement encrypting passwords
-    return password;
-  }
-
   private sanitizeOutput(user: User): SanitizedUser {
     return exclude(user, ['id', 'email_nonce', 'password', 'createdAt', 'updatedAt']);
   }
 
+  private async passwd_encrypt(password: string): Promise<string> {
+    const saltOrRounds = await bcrypt.genSalt(10);
+    const hash = await bcrypt.hash(password, saltOrRounds);
+    return hash;
+  }
+
   async create(user: CreateUserDto): Promise<SanitizedUser> {
     return this.sanitizeOutput(await this.prisma.user.create({
       data: {
         username: user.username,
         email: user.email,
-        password: this.passwd_encrypt(user.password),
-        isAdmin: false,
+        password: await this.passwd_encrypt(user.password),
+        isAdmin: true,
       }
     }));
   }
 
+  async validateEmail(email: string): Promise<boolean> {
+    // Verify if email form is valid
+    const expression: RegExp = EMAIL_REGEX;
+    const isValidRegex: boolean = expression.test(email);
+
+    if (isValidRegex) {
+        // Check that email address doesn't already exist in the db
+        const user = await this.findOne({email});
+        return !user;
+    }
+    return false;
+  }
+
+  async validateUsername(username: string): Promise<boolean> {
+    // Verify if username form is valid
+    const expression: RegExp = USERNAME_REGEX;
+    const isValidRegex: boolean = expression.test(username);
+
+    if (isValidRegex) {
+        // Check that username doesn't already exist in the db
+        const user = await this.findOne({username});
+        return !user;
+    }
+    return false;
+  }
+
+  async validatePassword(password: string): Promise<boolean> {
+    const expression: RegExp = PASSWORD_REGEX;
+    return(expression.test(password));
+  }
```

Comment:
This could be handled in the DTOs using NestJS's `class-validator` and the `whitelist` property in the `main.ts` file. See [Auto-validation -  NestJS Documentation](https://docs.nestjs.com/techniques/validation#auto-validation)

Reply:
Noted. I'll add a validation pipeline.